[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/miguepoloc/toma-decisiones-mcda/blob/main/05_promethee_iot_palmor.ipynb)

# PROMETHEE II, caso IoT/WSN Palmor

Elección de tecnología de comunicación (**LoRaWAN, GSM/GPRS, Sigfox, Zigbee**) para una red de sensores IoT/WSN de monitoreo agroclimático en Palmor, corregimiento de Ciénaga (Sierra Nevada de Santa Marta, Magdalena), zona con conectividad limitada verificada vía MinTIC. Los 4 criterios (Alcance de comunicación, Autonomía de batería, Infraestructura/cobertura comercial en Colombia, Madurez/viabilidad comercial del proveedor) salieron del panel de evidencia de la Sesión 1. Los valores técnicos de la matriz de decisión son reales, verificados vía WebSearch (datasheets SIMCom/DigiKey, The Things Network, Lauridsen et al. 2019 *Sensors*/MDPI, noticias de apagado de 2G en Colombia). Mismos datos y pesos que TOPSIS/VIKOR/ELECTRE.

**Nota de alcance:** igual que ELECTRE arriba, esta aplicación de PROMETHEE al caso IoT/Palmor es una extensión nueva de este repositorio (en el curso real, S4 sigue con PROMETHEE sobre el caso cacao). Datos reales, método y convención de umbrales de la Sesión 4, resultado no visto aún en clase.

In [1]:
import numpy as np
from pyDecision.algorithm import promethee_ii

tecnologias = ["LoRaWAN", "GSM/GPRS", "Sigfox", "Zigbee"]
dataset = np.array([
    [10,   8,   2, 5],
    [10.5, 0.5, 3, 2],
    [40,   2,   5, 2],
    [0.07, 1.5, 2, 4],
], dtype=float)

# Las 4 son de beneficio (a diferencia del caso cacao), no hace falta negar ninguna columna
rangos = dataset.max(axis=0) - dataset.min(axis=0)
pesos = [0.16, 0.25, 0.488, 0.102]
Q = [0, 0, 0, 0]
S = [0, 0, 0, 0]
P = list(rangos)
F = ['t3', 't3', 't3', 't3']  # función lineal simple

## Flujos y ranking

In [2]:
flujo = promethee_ii(dataset, W=pesos, Q=Q, S=S, P=P, F=F,
                      sort=False, topn=0, graph=False, verbose=False)

print("Ranking final PROMETHEE II:")
for t, fila in sorted(zip(tecnologias, flujo), key=lambda x: -x[1][1]):
    print(f"  {t:10s} phi_neto = {fila[1]:+.3f}")

Ranking final PROMETHEE II:
  Sigfox     phi_neto = +0.465
  LoRaWAN    phi_neto = +0.057
  GSM/GPRS   phi_neto = -0.193
  Zigbee     phi_neto = -0.330


**Resultado:** 1º Sigfox (+0.465) · 2º LoRaWAN (+0.057) · 3º GSM/GPRS (-0.193) · 4º Zigbee (-0.330), reproduce el mismo orden que AHP y TOPSIS.